src/katsdpcalproc/pointing.py

```diff
@@ -345,56 +338,67 @@ def calc_pointing_offsets(session, beams, target, middle_time,
           - rough uncertainty (standard deviation) of (az, el) adjustment.
 
     """
+
+    if isinstance(existing_az_el_adjust, type(None)):
+        existing_az_el_adjust = np.zeros((len(ants), 2))
+    if middle_time < 16000000:
+        raise NotUnixTime("Middle times must be in unix time format")
+    if not isinstance(target, katpoint.Target):
+        raise NotKatpointTarget("Not a katpoint target object")
+
     pointing_offsets = {}
     # Iterate over receptors
-    for ant in sorted(session.observers):
+    for ant in sorted(ants):
         beams_freq = beams.get(ant.name, [])
+
         beams_freq = [b for b in beams_freq if b is not None and b.is_valid]
-        if not beams_freq:
-            user_logger.debug("%s had no valid primary beam fitted", ant.name)
-            continue
+
         offsets_freq = np.array([b.center for b in beams_freq])
         offsets_freq_std = np.array([b.std_center for b in beams_freq])
+
         weights_freq = 1. / offsets_freq_std ** 2
-        # Do weighted average of offsets over frequency chunks
-        results = np.average(offsets_freq, axis=0, weights=weights_freq,
-                             returned=True)
+        results = np.average(offsets_freq, axis=0, weights=weights_freq, returned=True)
+
         pointing_offset = results[0]
         pointing_offset_std = np.sqrt(1. / results[1])
-        user_logger.debug("%s x=%+7.2f'+-%.2f\" y=%+7.2f'+-%.2f\"", ant.name,
-                          pointing_offset[0] * 60, pointing_offset_std[0] * 3600,
-                          pointing_offset[1] * 60, pointing_offset_std[1] * 3600)
         # Get existing pointing adjustment
-        receptor = getattr(session.kat, ant.name)
-        az_adjust = receptor.sensor.pos_adjust_pointm_azim.get_value()
-        el_adjust = receptor.sensor.pos_adjust_pointm_elev.get_value()
+        az_adjust = existing_az_el_adjust[ants.index(ant)][0]
+        el_adjust = existing_az_el_adjust[ants.index(ant)][1]
         existing_adjustment = deg2rad(np.array((az_adjust, el_adjust)))
         # Start with requested (az, el) coordinates, as they apply
         # at the middle time for a moving target
         requested_azel = target.azel(timestamp=middle_time, antenna=ant)
         # Correct for refraction, which becomes the requested value
         # at input of pointing model
         rc = RefractionCorrection()
+
         def refract(az, el):  # noqa: E306, E301
             """Apply refraction correction as at the middle of scan."""
             return [az, rc.apply(el, temperature, pressure, humidity)]
         refracted_azel = np.array(refract(*requested_azel))
+
```

Comment:
The general style in Python is to use blank lines sparingly. From [PEP 8](https://peps.python.org/pep-0008/#blank-lines):
```
Use blank lines in functions, sparingly, to indicate logical sections.
```